### DISCUSSION TITLE
how to hide a node within a mindmap diagram

### DISCUSSION THREAD

**Discussion by nnako** [plantuml/plantuml#1014] [Q&A] [answered]
Using plantuml's mindmap diagram, I want to hide specific nodes. Of course, I could omit them completely, but I want to render them hidden / visible whenever I like to by just changing a style setting. For such an implementation I thought of this:

```
@startmindmap
<style>
leafNode {
  .leaf_node_style {
    LineColor blue
    LineThickness 3.0
    BackgroundColor gold
    MinimumWidth 400
    Shadowing 6
}
}
node {
  .invisible {
    MaximumWidth 0
}
}
</style>
* root node text
** / <<invisible>>
*** some child node
*** another child node
**** / <<invisible>>
***** a leaf node <<leaf_node_style>>
@endmindmap
```

The generated mindmap displays the `leaf_node_style` as defined, but still shows the small nodes containing the slash "/" character, although the `invisible` style setting was used, defined at the top of the text:

![image](https://user-images.githubusercontent.com/5845602/167613843-626e266a-510b-47df-80c5-9799f229d244.png)

Is it possible to hide specific mindmap nodes so that the output does not show them anymore but instead, moves all it's child nodes to it's own position? Like so?:

![image](https://user-images.githubusercontent.com/5845602/167614420-b27d3599-f08c-4f2d-8936-27cdeee7f7ae.png)

**Comment by The-Lum** [comment, marked as answer]
Hello @nnako,

Here is a first attempt:

```puml
@startmindmap
<style>
leafNode {
  .leaf_node_style {
    LineColor blue
    LineThickness 3.0
    BackgroundColor gold
    MinimumWidth 400
    Shadowing 6
}
}
node {
  .invisible {
    MaximumWidth 0
    BackgroundColor transparent
    LineColor transparent
    Fontcolor transparent
}
}
</style>
* root node text
** / <<invisible>>
*** some child node
*** another child node
**** / <<invisible>>
***** a leaf node <<leaf_node_style>>
@endmindmap
```

>[![](https://www.plantuml.com/plantuml/svg/RP71QiCm38RlUGgTSsWB6pk5KZROQTrbWnsB6skng2qLMzawnjvzZbjIC2Kd_VydyY5lafgeWTW6Ssowf3yUrur7y_ag5k6t0LXCquxpk2jr2G5UYF59l4JO-n7dwDrHVs1C2UuMgv8-clumH1dPdl51l2t5bfZ263x8geFxrPb-SyRAD_403yrV_dZM8FwYHFjPO6jEr_NRlz9eE1rDHDRAkMwUXRMlcoBGBIz7QI6AA1GXnPCsRGjBwBhPQvsPd2K92BqZRmjRCiEY3cCLtzoVQ9XENh3SL-VFr0RPNfxi7m00)](https://www.plantuml.com/plantuml/uml/RP71QiCm38RlUGgTSsWB6pk5KZROQTrbWnsB6skng2qLMzawnjvzZbjIC2Kd_VydyY5lafgeWTW6Ssowf3yUrur7y_ag5k6t0LXCquxpk2jr2G5UYF59l4JO-n7dwDrHVs1C2UuMgv8-clumH1dPdl51l2t5bfZ263x8geFxrPb-SyRAD_403yrV_dZM8FwYHFjPO6jEr_NRlz9eE1rDHDRAkMwUXRMlcoBGBIz7QI6AA1GXnPCsRGjBwBhPQvsPd2K92BqZRmjRCiEY3cCLtzoVQ9XENh3SL-VFr0RPNfxi7m00)

And a second... with `padding 0`:

```puml
@startmindmap
<style>
leafNode {
  .leaf_node_style {
    LineColor blue
    LineThickness 3.0
    BackgroundColor gold
    MinimumWidth 400
    Shadowing 6
}
}
node {
  .invisible {
    padding 0
    BackgroundColor transparent
    LineColor transparent
    Fontcolor transparent
}
}
</style>
* root node text
** / <<invisible>>
*** some child node
*** another child node
**** / <<invisible>>
***** a leaf node <<leaf_node_style>>
@endmindmap
```
>[![](https://www.plantuml.com/plantuml/svg/RP4zQyD038Pt_mhD6P90IoSJGWkTsYujT0mNIxL5pfAva_j1wN_lMGaUJF2avtsCNdJxR25Ppu9z6Aewsq-aNHKfVBme4lnM0EjfEaWP3nux17XYeGUDck0OHvhHMyVDIIXdk5bldTw7vjGc7GNFUgiHFNXcuNxitncjWzljsNxj0keNImjtrL_vPAx1yicPZtE18I1EtlKjbeBa8IGIMzHT9eygrYmJtrrlBlTOGL8ry2v6trQjLh21kfuhxOfJMDQUeEauehlEWgXrb1RuslzkmtHjrqg-k7onzYHuUQr_)](https://www.plantuml.com/plantuml/uml/RP4zQyD038Pt_mhD6P90IoSJGWkTsYujT0mNIxL5pfAva_j1wN_lMGaUJF2avtsCNdJxR25Ppu9z6Aewsq-aNHKfVBme4lnM0EjfEaWP3nux17XYeGUDck0OHvhHMyVDIIXdk5bldTw7vjGc7GNFUgiHFNXcuNxitncjWzljsNxj0keNImjtrL_vPAx1yicPZtE18I1EtlKjbeBa8IGIMzHT9eygrYmJtrrlBlTOGL8ry2v6trQjLh21kfuhxOfJMDQUeEauehlEWgXrb1RuslzkmtHjrqg-k7onzYHuUQr_)


_If that can help,_
Regards.

**Reply by nnako** [reply]
hi @The-Lum .

Thanks for your answer.

As you see in the pictures above, there is always at least this little gap when making the node "invisible" inbetween. When using dashed connection "arrows" by utilizing `skinparam ArrowStyle 10`, even this gap becomes less visible:

```
@startmindmap
skinparam ArrowStyle 10
<style>
leafNode {
  .leaf_node_style {
    LineColor blue
    LineThickness 3.0
    BackgroundColor gold
    MinimumWidth 400
    Shadowing 6
}
}
node {
  .invisible {
    padding 0
    BackgroundColor transparent
    LineColor transparent
    Fontcolor transparent
}
}
</style>
* root node text
** / <<invisible>>
*** some child node
*** another child node
**** / <<invisible>>
***** a leaf node <<leaf_node_style>>
@endmindmap
```

![image](https://user-images.githubusercontent.com/5845602/168489038-5a985d04-22f9-42b8-914a-eba4c1236f8c.png)

**Reply by The-Lum** [reply]
Hello @nnako, and all,

Good thing, and if you want to do that by `style`, we can just add on style:

```puml
arrow {
  Linestyle 10
}
```
>[![](https://www.plantuml.com/plantuml/svg/RP4nJyD038Nt-nMUExH58AQega1Y0XQG6AjhpZHMBtPqvr0Gwd_dubGPGaU_zrb-yjjczKbRuj3wpbLPVo9kN4J_-Ie1uTS1B8TfnsNScMqYm3CnFagK1FlOuoIzDrGV6NE6s-NQr0TV7mz9UWuZVf0Op7WXfhPlFoXe0tVhaNvhV90JyG7kotnsPyTJ4E8lohIV8dG-X86yVaUJvzpvXAopm7FdILZh_uvTzod9QRXd--C3RjR5g5QNPoqWYIXOJCLlTOi5hA2gfhIRmXGjIujGDnI3iQPv5cqmpUHh-qR3K8LXnP_LKgWjShXK-GS0)](https://www.plantuml.com/plantuml/uml/RP4nJyD038Nt-nMUExH58AQega1Y0XQG6AjhpZHMBtPqvr0Gwd_dubGPGaU_zrb-yjjczKbRuj3wpbLPVo9kN4J_-Ie1uTS1B8TfnsNScMqYm3CnFagK1FlOuoIzDrGV6NE6s-NQr0TV7mz9UWuZVf0Op7WXfhPlFoXe0tVhaNvhV90JyG7kotnsPyTJ4E8lohIV8dG-X86yVaUJvzpvXAopm7FdILZh_uvTzod9QRXd--C3RjR5g5QNPoqWYIXOJCLlTOi5hA2gfhIRmXGjIujGDnI3iQPv5cqmpUHh-qR3K8LXnP_LKgWjShXK-GS0)


Regards.
